Common/include/geometry/elements/CElement.hpp

```diff
@@ -256,6 +270,51 @@ class CElement {
     for (unsigned short iDim = 0; iDim < nDim; iDim++) Kt_a(nodeA, iDim) += val_Kt_a[iDim];
   }
 
+  /*!
+   * \brief Set the current temperature of the element.
+   * \param[in] val_temperature - Value of the temperature to set.
+   */
+  inline void SetTemperature(const su2double val_temperature) { temperature = val_temperature; }
+
+  /*!
+   * \brief Get the current temperature of the element.
+   * \return Current temperature.
+   */
+
+  inline su2double GetTemperature(void) const { return temperature; }
+  
+  /*!
+   * \brief Set the temperature at a specific node.
+   * \param[in] iNode - Index of the node.
+   * \param[in] val_temperature - Temperature value to set at the node.
+   */
+inline void SetNodalTemperature(unsigned short iNode, su2double val_temperature) {
+  if (iNode < nodal_temperatures.size()) {
+    nodal_temperatures[iNode] = val_temperature;
+  }
+}
+
+/*!
+   * \brief Get the temperature at a specific node.
+   * \param[in] iNode - Index of the node.
+   * \return Temperature value at the specified node.
+   */
+inline su2double GetNodalTemperature(unsigned short iNode) const {
+  return (iNode < nodal_temperatures.size()) ? nodal_temperatures[iNode] : 0.0;
+}
+
+/*!
+   * \brief Compute the average temperature of the element from its nodal temperatures.
+   * \return Average temperature of the element.
+   */
+inline su2double ComputeAverageTemperature(void) const {
+  su2double sum_temperature = 0.0;
+  for (const auto &temp : nodal_temperatures) {
+    sum_temperature += temp;
+  }
+  return nodal_temperatures.empty() ? 0.0 : (sum_temperature / nodal_temperatures.size());
+}
```

Comment:
Here it should be the temperature interpolated at the Gaussian integration points.